src/main/java/me/karakelley/http/RequestValidator.java

```diff
@@ -0,0 +1,41 @@
+package me.karakelley.http;
+
+import me.karakelley.http.authorization.Authorizer;
+import me.karakelley.http.exceptions.InvalidRequestException;
+import me.karakelley.http.exceptions.UnauthorizedException;
+import java.util.Map;
+
+
+public class RequestValidator {
+  private Authorizer authorizer;
+
+  public RequestValidator(Authorizer authorizer) {
+    this.authorizer = authorizer;
+  }
+
+  public String validPath(String path) {
+    if (path == null) throw new InvalidRequestException("Invalid Path!");
+    return path;
+  }
+
+  public String validProtocol(String protocol) {
+    if (!protocol.equals("HTTP/1.1")) throw new InvalidRequestException("Invalid Protocol!");
+    return protocol;
+  }
+
+  public boolean requestIsAuthorized(HttpMethod method, Map<String, String> headers) {
+    String credentials = headers.getOrDefault("Authorization", null);
```

Comment:
nit pick: this could be inside the `if` statement, since you don't need to check this for `GET` requests.

Reply:
Additionally, there's an opportunity to simplify the authorizer here. This `getOrDefault("Authorization", null)` means that `String credentials` could be `null`, which you're then passing to the authorizer, which has to do a null check. There are a couple alternative options. You could use Optionals in some way (not sure what Java gives you for this, but fyi there are some languages where accessing a key from a hash will *always* return an Optional). If you don't want to get that intense, you could modify `checkAuthorization` to do something like:
```
if (credentials != null && authorizer.isAuthorized(credentials)) {...}
```